contracts/oracles/TwapOracle.sol

```diff
@@ -196,30 +192,47 @@ contract TwapOracle is OwnableUpgradeable, TwapInterface {
     }
 
     /**
-     * @notice Update new and old observations of lagging window if period elapsed.
+     * @notice Append current Observation and pick equal or just greater than window start timestamp,
+     * The observation which we are using except that we will delete rest of the observationn which are prior to this.
```

Comment:
We have some custom tags that we use to auto generate complete docs. For events we use `@custom:event` Can we add the nat spec here for the `TwapWindowUpdated` event?